CMakeLists.txt

```diff
@@ -47,43 +53,22 @@ PROJECT(cuda-api-wrappers
 # General C++ build settings
 # ----------------------------
 
+# Here we set a PRIVATE compile option for all targets in this project
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wpedantic" )
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
 
-include_directories( "src/" )
+set(CMAKE_CXX_EXTENSIONS OFF)
 
 # -------------
 # CUDA
 # -------------
-
-#find_package(CUDA 7.0 REQUIRED) # Why do I to do this damn it ?!
-#include_directories( "${CUDA_TOOLKIT_INCLUDE}" )
-include_directories( ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES} )
 include(HandleCUDAComputeCapability)
 
+# Here we set a PRIVATE compile option for all targets in this project
 set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xcompiler -Wall" )
-set(CMAKE_CUDA_STANDARD 11)
```

Comment:
What's the rationale for removing this?

Reply:
To modern CMake one should rather use `target` based commands, thus I added `set_target_properties(cuda-api-wrappers PROPERTIES CUDA_STANDARD 11)` sadly there is **no** `target_compile_features(cuda-api-wrappers PUBLIC cuda_std_11)` command possible. This would pass the necessity of a CUDA standard to users of the library. But I added `target_compile_features(cuda-api-wrappers PUBLIC cxx_std_11)`, since the static build library `libcuda-api-wrappers.a` is only built by C++ source files, and setting `cxx_std_11` as a `PUBLIC` compile feature results now in `nvcc -std=c++11 ...` for CUDA projects, which use `cuda-api-wrappers`. Seems to me a bit non-transparent, but at least the standard seems to be exported, which is the important point here. As far as I tested it `set(CMAKE_CUDA_STANDARD 11)` does not export the CUDA standard.

Reply:
1. Wouldn't this only apply to _this_ `CMakeFile.txt`'s targets? 
2. Targets have a CUDA_STANDARD property, and this sets it - [the docs say so](https://cmake.org/cmake/help/v3.8/variable/CMAKE_CUDA_STANDARD.html) (beginning with v3.8).

Reply:
`set(CMAKE_CUDA_STANDARD 11)` applies to all targets in this `CMakeLists.txt`, whereas
`target_compile_features(cuda-api-wrappers PUBLIC cxx_std_11)` also applies to targets, which link against `cuda-api-wrappers`, which is reasonable, since the headers use c++11 features, I guess.

Reply:
Fair enough, but there's still no reason to remove the file-level setting. When `cuda_std_11` becomes possibly (and the relevant CMake version popular enough) we'll add that too.